container_linux.go

```diff
@@ -777,3 +815,37 @@ func (c *linuxContainer) currentState() (*State, error) {
 	}
 	return state, nil
 }
+
+// orderNamespacePaths sorts that namespace paths into a list of paths that we
+// can safely setns to.
+func orderNamespacePaths(namespaces map[configs.NamespaceType]string) ([]string, error) {
+	paths := []string{}
+	for _, nsType := range []configs.NamespaceType{
+		// TODO: enable configs.NEWUSER,
+		configs.NEWIPC,
+		configs.NEWUTS,
+		configs.NEWNET,
+		configs.NEWPID,
+		// mnt namespace must be the last one. After switching to mnt
+		// namespace, the old /proc will be inaccessible.
```

Comment:
added. Joining user namespace will only be applicable for execin process for now. There are some trouble with EPERM when setting up mnt/net when i was trying to let an init process join another init process in test. I didnt debug much further.

I figure that joining pid namespace is the most obvious usecase now. I can look into other usecase after (mnt/userns) if they make sense.